File: src/main/java/logic/entity/Address.java
package logic.entity;

public class Address {
    private int id;
    private String countryName;
    private String cityName;
    private String streetName;
    private Integer houseNumber;
    private Integer flatNumber;
    private Integer index;

    public Address() {
    }

    public int getId() {
        return this.id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCountryName() {
        return this.countryName;
    }

    public void setCountryName(String countryName) {
        this.countryName = countryName;
    }

    public String getCityName() {
        return this.cityName;
    }

    public void setCityName(String cityName) {
        this.cityName = cityName;
    }

    public String getStreetName() {
        return this.streetName;
    }

    public void setStreetName(String streetName) {
        this.streetName = streetName;
    }

    public Integer getHouseNumber() {
        return this.houseNumber==0 ? null:houseNumber;
    }

    public void setHouseNumber(Integer houseNumber) {
        this.houseNumber = houseNumber;
    }

    public Integer getFlatNumber() {
        return this.flatNumber==0 ? null:flatNumber;
    }

    public void setFlatNumber(Integer flatNumber) {
        this.flatNumber = flatNumber;
    }

    public Integer getIndex() {
        return this.index==0 ? null:index;
    }

    public void setIndex(Integer index) {
        this.index = index;
    }


    public String toString() {
        StringBuilder result = new StringBuilder();
        if(countryName!=null){
            result.append(countryName);
        }
        if(cityName!=null){
            result.append(" г.").append(cityName);
        }
        if(streetName!=null){
            result.append(" ул.").append(streetName);
        }
        if(houseNumber!=null){
            result.append(" д.").append(houseNumber);
        }
        if(flatNumber!=null){
            result.append(" кв.").append(flatNumber);
        }

        return result.toString();
    }

}

File: src/main/webapp/web/js/entity/address.js
function Address() {
    this._id = undefined;
    this._countryName = undefined;
    this._cityName = undefined;
    this._streetName = undefined;
    this._houseNumber = undefined;
    this._flatNumber = undefined;
    this._index = undefined;
}
Address.prototype = {
    setId : function (id) {
        this._id = id;
    },
    getId : function () {
        return this._id;
    }
};
Address.prototype = {
    setCountryName : function (countryName) {
        this._countryName = countryName;
    },
    getCountryName : function () {
        return this._countryName;
    }
};
Address.prototype = {
    setCityName : function (cityName) {
        this._cityName = cityName;
    },
    getCityName : function () {
        return this._cityName;
    }
};
Address.prototype = {
    setStreetName : function (streetName) {
        this._streetName = streetName;
    },
    getStreetName : function () {
        return this._streetName;
    }
};
Address.prototype = {
    setHouseNumber : function (houseNumber) {
        this._houseNumber = houseNumber;
    },
    getHouseNumber : function () {
        return this._houseNumber;
    }
};
Address.prototype = {
    setFlatNumber : function (flatNumber) {
        this._flatNumber = flatNumber;
    },
    getFlatNumber : function () {
        return this._flatNumber;
    }
};
Address.prototype = {
    setIndex : function (index) {
        this._index = index;
    },
    getIndex : function () {
        return this._index;
    }
};